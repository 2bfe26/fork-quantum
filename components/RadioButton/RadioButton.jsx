import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Colors from '../Colors';
import theme from '../../theme';

const Wrapper = styled.span`
  background-color: ${Colors.NEUTRAL.GRAY.WHITE};
  border: 1px solid ${Colors.PRIMARY.BLUE.PEACOCK};
  border-radius: ${theme.sizes.radius};
  box-sizing: border-box;
  cursor: pointer;
  display: inline-block;
  font-weight: 500;
  height: ${theme.sizes.fieldHeight};
  margin-right: 20px;
  padding: 10px 30px;
  position: relative;
  overflow: hidden;
  text-align: center;
  ${theme.mixins.transition()};

  &:last-child {
    margin-right: 0;
  }

  label {
    color: ${Colors.PRIMARY.BLUE.PEACOCK};
    cursor: pointer;
    position: relative;
  }

  input {
    appearance: none;
    background-color: ${Colors.NEUTRAL.GRAY.WHITE};
    cursor: pointer;
    height: 100%;
    left: 0;
    margin: 0;
    position: absolute;
    top: 0;
    width: 100%;
    ${theme.mixins.transition()};

    &:checked {
      background-color: ${Colors.PRIMARY.BLUE.PEACOCK};
      ${theme.mixins.transition()};

      ~ label {
        color: ${Colors.NEUTRAL.GRAY.WHITE};
        ${theme.mixins.transition()};
      }
  }
`;

const RadioButton = ({
  children,
  id,
  value,
  name,
}) => (
  <Wrapper>
    <input type="radio" id={id} value={value} name={name} />
    <label htmlFor={id}>{children}</label>
  </Wrapper>
);

RadioButton.propTypes = {
  children: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default RadioButton;
