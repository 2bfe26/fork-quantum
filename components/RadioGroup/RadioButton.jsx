import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ButtonGroupLabel from '../shared/ButtonGroupLabel';
import HiddenInput from '../shared/HiddenInput';
import Icon from '../Icon';
import uniqId from '../shared/uniqId';

const ID_GENERATOR = uniqId('radio-button-');

const LockIcon = styled(Icon).attrs({
  name: 'lock',
})``;

const ButtonIcon = styled(Icon)`
  margin-right: 8px;
`;

const Wrapper = styled.div`
  ${({ inline }) =>
    inline &&
    `
    display: inline-block;
    margin-right: 16px;
    vertical-align: top;

    :last-child {
      margin-right: 0;
    }
  `}

  ${LockIcon} {
    font-size: 17px;
    margin-left: 8px;
  }
`;

class Radio extends React.Component {
  constructor(props) {
    super(props);

    const { id } = props;

    this._id = id || ID_GENERATOR.next().value;
  }

  render() {
    const {
      children,
      label,
      error,
      disabled,
      onChange,
      value,
      checked,
      icon,
      id,
      inline,
      ...rest
    } = this.props;
    const skin = checked ? 'primary' : 'secondary';

    return (
      <Wrapper inline={inline}>
        <HiddenInput
          type="radio"
          checked={checked}
          disabled={disabled}
          id={this._id}
          onChange={e => onChange({ value, label }, e)}
          value={value}
          skin={skin}
          error={error}
          {...rest}
        />
        <ButtonGroupLabel
          checked={checked}
          disabled={disabled}
          error={error}
          htmlFor={this._id}
          skin={skin}
        >
          {icon && <ButtonIcon name={icon} />}
          {children || label}
          {disabled && <LockIcon />}
        </ButtonGroupLabel>
      </Wrapper>
    );
  }
}

Radio.displayName = 'RadioGroup.Button';

Radio.defaultProps = {
  checked: false,
  children: undefined,
  disabled: false,
  error: false,
  icon: undefined,
  id: undefined,
  inline: false,
  label: undefined,
  onChange: () => {},
};

Radio.propTypes = {
  checked: PropTypes.bool,
  children: PropTypes.string,
  disabled: PropTypes.bool,
  error: PropTypes.bool,
  icon: PropTypes.string,
  id: PropTypes.string,
  inline: PropTypes.bool,
  label: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string.isRequired,
};

export default Radio;
