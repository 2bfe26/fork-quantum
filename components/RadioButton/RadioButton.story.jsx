import React from 'react';
import { storiesOf } from '@storybook/react';
import HowToImport from '../../.storybook/decorators/HowToImport';
import RadioButton from './RadioButton';

const stories = storiesOf('4. Forms', module);
stories
  .addDecorator(HowToImport)
  .add('RadioButton', () => (
    <div>
      <RadioButton id="radio" value="radio" name="radio" label="RadioButton" />
      <RadioButton id="radioTwo" value="radioTwo" name="radio" label="Radio Button 2" />
    </div>
  ));
