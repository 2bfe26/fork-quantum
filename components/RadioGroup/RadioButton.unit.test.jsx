import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from './RadioButton';

describe('<RadioGroup.Radio />', () => {
  describe('snapshot', () => {
    it('simple', () => {
      const component = <RadioButton value="Foo" label="Foo" />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('checked', () => {
      const component = <RadioButton value="Foo" label="Foo" checked />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('inline', () => {
      const component = <RadioButton value="Foo" label="Foo" inline />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('icon', () => {
      const component = <RadioButton value="Foo" label="Foo" icon="check" />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('disabled', () => {
      const component = <RadioButton value="Foo" label="Foo" disabled />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('error', () => {
      const component = <RadioButton value="Foo" label="Foo" error />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('disabled with an error', () => {
      const component = <RadioButton value="Foo" label="Foo" error disabled />;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('children', () => {
      const component = <RadioButton value="Foo">Foo</RadioButton>;
      expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });
  });
});
