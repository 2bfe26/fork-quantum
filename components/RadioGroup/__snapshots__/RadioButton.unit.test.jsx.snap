// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup.Radio /> snapshot checked 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={true}
    className="c1"
    disabled={false}
    id="radio-button-1"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={true}
    className="c2"
    disabled={false}
    htmlFor="radio-button-1"
  >
    Foo
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot children 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    id="radio-button-7"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={false}
    htmlFor="radio-button-7"
  >
    Foo
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot disabled 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: rgba(204,204,204,0.4);
  border: 1.5px solid #999999;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #999999;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

.c4 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={true}
    id="radio-button-4"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={true}
    htmlFor="radio-button-4"
  >
    Foo
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c3 "
      style={
        Object {
          "color": "",
        }
      }
    >
      lock
    </span>
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot disabled with an error 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: rgba(204,204,204,0.4);
  border: 1.5px solid #999999;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #999999;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

.c4 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={true}
    id="radio-button-6"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={true}
    htmlFor="radio-button-6"
  >
    Foo
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c3 "
      style={
        Object {
          "color": "",
        }
      }
    >
      lock
    </span>
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot error 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #ff2d4d;
  border-color: #ff2d4d;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    id="radio-button-5"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={false}
    htmlFor="radio-button-5"
  >
    Foo
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot icon 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c3 {
  margin-right: 8px;
}

.c0 .c4 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    id="radio-button-3"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={false}
    htmlFor="radio-button-3"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c3"
      style={
        Object {
          "color": "",
        }
      }
    >
      check
    </span>
    Foo
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot inline 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  display: inline-block;
  margin-right: 16px;
  vertical-align: top;
}

.c0:last-child {
  margin-right: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    id="radio-button-2"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={false}
    htmlFor="radio-button-2"
  >
    Foo
  </label>
</div>
`;

exports[`<RadioGroup.Radio /> snapshot simple 1`] = `
.c2 {
  display: block;
  margin-bottom: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  margin-bottom: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c1 {
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 .c3 {
  font-size: 17px;
  margin-left: 8px;
}

<div
  className="c0"
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    id="radio-button-0"
    onChange={[Function]}
    type="radio"
    value="Foo"
  />
  <label
    checked={false}
    className="c2"
    disabled={false}
    htmlFor="radio-button-0"
  >
    Foo
  </label>
</div>
`;
